### DISCUSSION TITLE
Add a defineRoute(route: RouteRecordRaw) API similar to definePageMeta()

### DISCUSSION THREAD

**Discussion by nurdism** [nuxt/nuxt#17999] [Ideas]
With new the router, there seem to be a lot of nice features that can't be accessed if deciding to use the nuxt pages. An idea could be to include a `defineRoute(route: RouteRecordRaw)` API that could *extend* the generated page routes.

An example would be  `pages/test/[test].ts` would generate the following routes config:

```json
{
    "name": "test-test",
    "path": "/test/:test?",
    "file": "src/pages/test/[test].vue",
    "children": []
}
```

If I where to include the following in the setup script:
```vue
<script lang="ts" setup>
  definePage({
    path: "/test/:test(\\d+)",
    strict: true,
    meta: { authed: true }
  })
</script>
```
it would produce the following:
```json
{
    "name": "test-test",
    "path": "/test/:test(\\d+)",
    "strict": true,
    "meta": { "authed": true },
    "file": "src/pages/test/[test].vue",
    "children": []
}
```

of course there should be some safeguards to some of the options like `children` & `file` so they can not be overridden. But a way to extend the route would be great for making custom path, enabling strict and setting meta.

**Comment by danielroe** [comment]
`definePageMeta` allows you to set route `meta`, and `strict` is an option provided directly to `createRouter`. So, just to clarify, this is mostly focused on setting [these properties](https://next.router.vuejs.org/api/#routerecordraw):

* `redirect`
* `alias`
* `name`
* `beforeEnter`
* `props`

It is also worth observing, for anyone else coming here, that the `pages:extend` hook gives a way to customise all of these properties on a module/project level.

Regardless of whether this is supported in the core framework, there would definitely be scope for a nuxt module (ala [@nuxtjs/router-extras](https://github.com/nuxt-community/router-extras-module/)) to add this feature.

**Reply by nurdism** [reply]
> strict is an option provided directly to createRouter.

Actually strict is an [undocumented](https://github.com/vuejs/router/issues/1277) option.
> The `strict` option allows to take into account trailing slashes and is disabled by default.

I just figured we have `definePageMeta()` witch is already doing the the same thing, but with the `meta` option for the routes, why not just have it mutate the route.

Using `pages:extend` does work, but it would be nice to be able to extend the route in the page component rather than adding/modifying a large amount of page routes in the `nuxt.config.ts`
